table-api/src/main/java/io/deephaven/api/ConcurrencyControl.java

```diff
@@ -36,4 +36,30 @@ public interface ConcurrencyControl<T> {
      * @return a new instance of T with serial concurrency control applied.
      */
     T withSerial();
+
+    /**
+     * Designates the expression as a barrier filter with the specified barrier name.
+     * <p>
+     * A barrier filter does not affect concurrency but imposes an ordering constraint on filters that respect the same
+     * barrier. Filters specified before the barrier will be evaluated before it, and those specified after will be
```

Comment:
Only if they also respect the barrier.

Reply:
You should define the rules for barriers.  Presumably they must correctly implement hashCode and equals.